docs/language/control-flow.md

```diff
@@ -112,6 +113,78 @@ if let number = noNumber {
 }
 ```
 
+## Switch
+
+Switch-statements compare a value against several possible values of the same type, in order.
+When an equal value is found, the associated block of code is executed.
+
+The switch-statement starts with the `switch` keyword, followed by the tested value,
+followed by the cases inside opening and closing braces.
+The test expression must be equatable.
+The braces are required and not optional.
+
+Each case is a separate branch of code execution
+and starts with the `case` keyword,
+followed by a possible value, a colon (`:`),
+and the block of code that should be executed
+if the case's value is equal to the tested value.
+
+The block of code associated with a switch case
+[does not implicitly fall through](#no-implicit-fallthrough),
+and must contain at least one statement.
+Empty blocks are invalid.
+
+An optional default case may be given by using the `default` keyword.
+The block of code of the default case is executed
+when none of the previous case tests succeeded.
+It must always appear last.
+
+```cadence
+fun word(_ n: Int): String {
+    // Test the value of the parameter `n`
+    switch n {
+    case 1:
+        // If the value of variable `n` is equal to `1`,
+        // then return the string "one"
+        return "one"
+    case 2:
+        // If the value of variable `n` is equal to `1`,
+        // then return the string "two"
+        return "two"
+    default:
+        // If the value of variable `n` is neither equal to `1` nor to `2`,
+        // then return the string "other"
+        return "other"
+    }
+}
+```
+
+### `break`
+
+The block of code associated with a switch case may contain a `break` statement.
+It ends the execution of the switch statement immediately
+and transfers control to the code after the switch statement
+
+### No Implicit Fallthrough
+
+Unlike switch statements in some other languages,
+switch statements in Cadence do not "fall through":
+execution of the switch statement finishes as soon as the block of code
+associated with the first matching case is completed.
+No explicit `break` statement is required.
```

Comment:
Do we allow duplicated cases? what would happen for the following case?
```
y = 1
switch x {
case 1:
  y = y + 1
case 1:
  y = y + 2
default:
  y = y + 3
}
```
If we allow, what's the result of `y`?

Reply:
Great question, I'll add that

Reply:
Described in 2b32abf9f1ce10e3ed836b12512c3173f75dae64